src/Solnet.Wallet/Utilities/BitWriter.cs

```diff
@@ -75,15 +75,15 @@ private static byte[] ToByteArray(BitArray bits)
             int arrayLength = bits.Length / 8;
             if (bits.Length % 8 != 0)
                 arrayLength++;
-            byte[] array = new byte[arrayLength];
+            Span<byte> array = stackalloc byte[arrayLength];
 
             for (int i = 0; i < bits.Length; i++)
             {
                 int b = i / 8;
                 int offset = i % 8;
                 array[b] |= bits.Get(i) ? (byte)(1 << offset) : (byte)0;
             }
-            return array;
+            return array.ToArray();

```

Comment:
Same thing here, the array is returned, so a new copy is created in the heap. Revert to prevent more CPU usage in the copy and pointless stack allocation.